src/dialog_functions.cpp

```diff
@@ -0,0 +1,96 @@
+#include <QCoreApplication>
+#include <QMessageBox>
+#include <QTimer>
+#include <functional>
+
+#include "dialog_functions.h"
+#include "networkmanager.h"
+
+void call_error(QString p_message)
+{
+  QMessageBox *msgBox = new QMessageBox;
+
+  msgBox->setAttribute(Qt::WA_DeleteOnClose);
+  msgBox->setText(QCoreApplication::translate("debug_functions", "Error: %1")
+                      .arg(p_message));
+  msgBox->setWindowTitle(
+      QCoreApplication::translate("debug_functions", "Error"));
+
+  // msgBox->setWindowModality(Qt::NonModal);
+  msgBox->exec();
+}
+
+void call_notice(QString p_message)
+{
+  QMessageBox *msgBox = new QMessageBox;
+
+  msgBox->setAttribute(Qt::WA_DeleteOnClose);
+  msgBox->setText(p_message);
+  msgBox->setWindowTitle(
+      QCoreApplication::translate("debug_functions", "Notice"));
+
+  msgBox->setStandardButtons(QMessageBox::NoButton);
+
+  QTimer::singleShot(3000, msgBox, std::bind(&QMessageBox::setStandardButtons,msgBox,QMessageBox::Ok));
+
+  msgBox->exec();  
+
+}
+
+LoginDialog::LoginDialog(QWidget *parent):QDialog(parent)
+{
+    setupUi();
+    setWindowTitle(QCoreApplication::translate("dialog_functions", "Login"));
+    setModal(true);
+}
+
+void LoginDialog::setupUi()
```

Comment:
For more complicated dialogs, let's use UIC, please. That way we don't have to write the UI in a C++ mess.